megamek/src/megamek/client/ui/swing/widget/GeneralInfoMapSet.java

```diff
@@ -356,15 +356,18 @@ public void setEntity(Entity en) {
             element.setString(""); //$NON-NLS-1$
         }
 
-        int i = 0;
         if ((null != en.getGame())
                 && en.getGame().getOptions().booleanOption(OptionsConstants.ADVANCED_STRATOPS_QUIRKS)) {
-            addOptionsToList(i, en.getQuirks(), quirksR);
+            addOptionsToList(en.getQuirks(), quirksR);
         }
         
+        // Note that, due to the fact that the quirks display is hardcoded to 40 labels in height,
+        // unless you're running at 1024x12000 resolution, you're never going to see partial repairs on the unit info card.
+        // That fix will require changing the data types of partRepsR and quirksR to a dynamically allocated list,
+        // which is not a very simple change, so I'm leaving it for another time.
```

Comment:
Seems like a case for a couple JScrollPanes so that the full lists can be displayed.

As for switching from arrays to lists, that shouldn't be too hard.  From what I looked at, it'd only be about 7 or 8 lines of code that would actually have to be changed.

Reply:
JScrollPane didn't quite work out, but I was finally able to figure out a solution to dynamically display a variable number of rows. So now we'll be seeing quirks and partial repairs on that unit card.